graphyte.py

```diff
@@ -169,12 +171,24 @@ def _thread_loop(self):
             if current_time - last_check_time >= self.interval:
                 last_check_time = current_time
                 if messages:
-                    self.send_socket(b''.join(messages))
+                    while messages:
+                        count_to_send = min(self.batch_size, len(messages))
+                        # Move batch from messages to batch 
+                        batch = messages[0:count_to_send]
+                        del messages[0:count_to_send]
+                        # Send batch
+                        self.send_socket(b''.join(batch))
                     messages = []
 
         # Send any final messages before exiting thread
         if messages:
-            self.send_socket(b''.join(messages))
+             while messages:
+                 count_to_send = min(self.batch_size, len(messages))
+                 # Move batch from messages to batch 
+                 batch = messages[0:count_to_send]
+                 del messages[0:count_to_send]
+                 # Send batch
+                 self.send_socket(b''.join(batch))
```

Comment:
Same as above, and again, with the for loop we can avoid the `if messages:` block.